### PR TITLE
docs: rewrite some docs for mdx v2

### PR DESCRIPTION
**Pull request by slorber**
## Motivation

Follow-up of the MDX v2 PR: https://github.com/facebook/docusaurus/pull/8288

Many docs pages had outdated content.

We probably can keep improving this new version but at least I tried to remove everything that was wrong/misleading for the upcoming Docusaurus version

## Test Plan

preview

### CODE REVIEW

File: website/docs/guides/markdown-features/markdown-features-react.mdx [hunk 1]
```diff
@@ -231,227 +242,25 @@ If you don't wrap your imported MDX with `MDXContent`, the global scope will not
...
-```
+Some valid CommonMark features won't work with MDX ([more info](https://mdxjs.com/docs/what-is-mdx/#markdown)), notably:
 
-```mdx-code-block
-</div>
-<div className={styles.wrappingBlock}>
-<BrowserWindow>
+- Indented code blocks: use triple backticks instead
+- Autolinks (`<http://localhost:3000>`): use regular link syntax instead (`[http://localhost:3000](http://localhost:3000)`)
+- HTML syntax (`<p style="color: red;>`): use JSX instead (`<p style="color: red;>`)
```
Comment (lebalz) [inline website/docs/guides/markdown-features/markdown-features-react.mdx:253]:
```suggestion
- HTML syntax (`<p style="color: red;>`): use JSX instead (`<p style={{color: 'red'}}>`)
```

File: website/docs/guides/markdown-features/markdown-features-react.mdx [hunk 2]
```diff
@@ -204,7 +215,7 @@ From MDX v2+ onward (Docusaurus v3+), lower-case tag names are always rendered a
 
 :::caution
 
-This feature is powered by [a wrapper provider](https://mdx-git-renovate-babel-monorepo-mdx.vercel.app/advanced/components#mdxprovider). If you are importing Markdown in a React page, you have to supply this provider yourself through the `MDXContent` theme component.
+This feature is powered by [a MDXProvider](https://mdxjs.com/docs/using-mdx/#mdx-provider). If you are importing Markdown in a React page, you have to supply this provider yourself through the `MDXContent` theme component.
```
Comment (Josh-Cena) [inline website/docs/guides/markdown-features/markdown-features-react.mdx:218]:
```suggestion
This feature is powered by [an `MDXProvider`](https://mdxjs.com/docs/using-mdx/#mdx-provider). If you are importing Markdown in a React page, you have to supply this provider yourself through the `MDXContent` theme component.
```

File: website/docs/guides/markdown-features/markdown-features-react.mdx [hunk 3]
```diff
@@ -17,14 +17,29 @@ import styles from './markdown-features-react.module.css';
 
 Docusaurus has built-in support for [MDX v2](https://mdxjs.com/), which allows you to write JSX within your Markdown files and render them as React components.
 
-:::note
+:::info MDX vs COMMONMARK
```
Comment (Josh-Cena) [inline website/docs/guides/markdown-features/markdown-features-react.mdx:20]:
```suggestion
:::info MDX vs. CommonMark
```

File: website/docs/guides/markdown-features/markdown-features-react.mdx (hunk 1 above)
Comment (slorber) [reply website/docs/guides/markdown-features/markdown-features-react.mdx:253]: Thanks :)

**Pull request merged**